EIPS/eip-3561.md

```diff
@@ -0,0 +1,59 @@
+---
+eip: 3561
+title: Trust Minimized Upgradeability Proxy
+author: <@SamPorter1984>
+discussions-to: https://ethereum-magicians.org/t/trust-minimized-proxy/5742
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-05-09
+---
+
+## Simple Summary
+Additional storage slots to decrease trust in interaction with upgradeable smart contracts.
+
+## Abstract
+Removing trust from upgradeability proxy is required for anonymous developers. To achieve that, disallowing sudden, potentially malicious upgrades is required. This EIP introduces a time lock, a period of one month, before defined by proxy admin implementation becomes active implementation.
+
+## Motivation
+It's usually not possible for anonymous developers who use upgradeability proxies to gain community trust.
+
+Fairer, better future for humanity absolutely requires some developers to stay anonymous while still attract vital attention to solutions they propose and at the same time leverage the benefits of possible upgradeability. For example, a project like Collateralized Asset Protocol allows to trade stocks and securities on the blockchain in a purely decentralized way making the developers of such projects potentially risking their freedom. Aletheo was born to fight tyranny.
+
+Extinction is more likely with closed internet borders. As long as internet has eyes everywhere, devastating conventional warfare is less likely to happen regardless of power vacuum or exponential increase in existential threats.
+
+## Specification
+
+All admin interactions with trust minimized proxy must emit an event to make admin actions trackable. 
+
+### Logic contract address
+
+`NEXT_IMPLEMENTATION_SLOT` stores the address of next logic to be implemented. Set by `proposeToAndCall(address implementation, bytes calldata data)` method.
```

Comment:
I think this specification currently fails to stand alone. One must read the implementation to find the values (such as that of `NEXT_IMPLEMENTATION_SLOT`) and understand what functions are required in the interface. I suggest to consider looking at some other ERCs which define interfaces.

Reply:
Sure, I can make it more detailed

Reply:
Should these changes suffice?